src/components/login/login.tsx

```diff
@@ -54,22 +88,102 @@ class Login extends React.Component<Props, DefaultState> {
         });
     };
 
+    public handlePassphraseChange = (element) => {
+        this.setState({
+            passphrase: element.target.value,
+        });
+    };
+
+    public handleFilePathChange = (element) => {
+        this.setState({
+            filePath: element.target.value,
+        });
+    };
+
     public handleAgreementChange = (element, data) => {
         this.setState({
             agreement: data.checked ? 1 : 0,
         });
     };
 
-    public handleLogin = () => {
-        if (this.state.agreement) {
-            this.props.login(this.state.publicKey, this.state.privateKey);
-            return;
+    public handleLogin = async () => {
+        logger.info("Logging in user..");
+
+        try {
+            // verify keypair-relation
+            const keyPair = await this.cryptoKeyUtil.verifyKeyPair(
+                this.state.publicKey,
+                this.state.privateKey,
+            );
+
+            if (this.state.agreement) {
+                this.props.login(this.state.publicKey, this.state.privateKey);
+                return;
+            }
+            toast.info("Please agree to the Terms and Conditions");
+            goToUrl("/profile");
+        } catch (err) {
+            logger.error(err);
+            toast.error("Please supply correct credentials!");
         }
-        toast.info("Please agree to the Terms and Conditions");
     };
 
+    public fileRead = (filePath) => {
+        logger.info("Reading local file..");
+        const contents = fs.readFileSync(filePath, "utf8");
+        this.setState({ filePath, open: true, encryptedString: contents });
+    };
+
+    public generateKeypair = () => {
+        logger.info("Generating new keypair..");
+        const keypair = this.cryptoKeyUtil.generateKeyPair();
+        const pubKey = keypair.getPublic(true, "hex");
+        const privKey = keypair.getPrivate("hex");
+
+        logger.info("PUBKEY: " + pubKey);
+        logger.info("PRIVKEY: " + privKey);
```

Comment:
Why are the generated keys logged? Please remove these logs if they are unnecessary.

Reply:
It was for debugging purposes, but indeed it is no longer necessary anymore.